packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1046,11 +1049,25 @@ export async function deleteDataset(node: api.IZoweTreeNode, datasetProvider: ap
                 ses.dirty = true;
             }
         });
-        datasetProvider.removeFavorite(node);
     } else {
         node.getSessionNode().dirty = true;
-        datasetProvider.removeFavorite(node);
     }
+    datasetProvider.removeFavorite(node);
+
+    const isMember = contextually.isDsMember(node);
+
+    // If the node is a dataset member, go up a level in the node tree
+    // to find the relevant, matching node
+    const nodeOfInterest = isMember ? node.getParent() : node;
+    const parentNode = fav
+        ? datasetProvider.findNonFavoritedNode(nodeOfInterest)
+        : datasetProvider.findFavoritedNode(nodeOfInterest);
```

Comment:
Can we call the `findEquivalentNode` method here to reduce duplication (same question as above)?